optimum/intel/openvino/modeling_visual_language.py

```diff
@@ -0,0 +1,1185 @@
+import logging
+import os
+import warnings
+from pathlib import Path
+from tempfile import TemporaryDirectory, gettempdir
+from typing import Dict, Optional, Tuple, Union
+
+import numpy as np
+import openvino as ov
+import torch
+from huggingface_hub import hf_hub_download
+from huggingface_hub.constants import HUGGINGFACE_HUB_CACHE
+from openvino._offline_transformations import apply_moc_transformations, compress_model_transformation
+from transformers import AutoConfig, GenerationConfig, GenerationMixin, PretrainedConfig
+from transformers.modeling_outputs import BaseModelOutputWithPooling
+
+from ...exporters.openvino import main_export
+from ...exporters.openvino.stateful import ensure_stateful_is_available
+from .configuration import OVConfig, OVWeightQuantizationConfig
+from .modeling_base import OVBaseModel
+from .modeling_decoder import CausalLMOutputWithPast, OVModelForCausalLM
+from .utils import OV_TO_PT_TYPE, _print_compiled_model_properties
+
+
+logger = logging.getLogger(__name__)
+
+core = ov.Core()
+
+
+class OVModelWithEmbedForCausalLM(OVModelForCausalLM):
+    def __init__(
+        self,
+        model: ov.Model,
+        text_embeds_model: ov.Model,
+        config: PretrainedConfig = None,
+        device: str = "CPU",
+        dynamic_shapes: bool = True,
+        ov_config: Optional[Dict[str, str]] = None,
+        model_save_dir: Optional[Union[str, Path, TemporaryDirectory]] = None,
+        quantization_config: Optional[Union[OVWeightQuantizationConfig, Dict]] = None,
+        **kwargs,
+    ):
+        self.model = model
+        self.text_emb_model = text_embeds_model
+        self.request = None
+        self.text_emb_request = None
+
+        super().__init__(
+            model, config, device, dynamic_shapes, ov_config, model_save_dir, quantization_config, **kwargs
+        )
+
+    def compile(self):
+        if self.request is None:
+            logger.info(f"Compiling the Language model to {self._device} ...")
+            self.request = core.compile_model(self.model, self._device, self.ov_config).create_infer_request()
+        self._compile_text_emb()
+
+    def _compile_text_emb(self):
+        if self.text_emb_request is None:
+            logger.info(f"Compiling the Text embeddings model to {self._device} ...")
+            self.text_emb_request = core.compile_model(self.text_emb_model, self._device, self.ov_config)
+
+    def to(self, device: str):
+        if isinstance(device, str):
+            self._device = device.upper()
+            self.clear_requests()
+
+        return self
```

Comment:
could make sense to modify `OVModel.to` to not have to overwrite it https://github.com/huggingface/optimum-intel/blob/9cc0ee41b5e9ef114afeaeb11ba715d4165b7d74/optimum/intel/openvino/modeling.py#L129 (just adding a `clear_requests` method), what do you think ?

Reply:
done